### ISSUE TITLE
Asserts on helloworld when using Fixed32

### ISSUE THREAD

**Issue by Marreck** [louis-langholtz/Box2D#49]
Hi,

I am trying to use the Fixed32 type but run into asserts when trying the hello world example. Specifically this one:

`Assertion failed: almost_equal(1, [&]() { return std::accumulate(std::begin(normalizedWeights), std
ts), RealNum(0)); }()), file C:\Users\Mark\Desktop\Box2dTest\Box2D/Collision/Simplex.hpp, line 288`

**Comment by louis-langholtz** [comment]
Thank you for the bug report!

Sorry about this problem you've encountered. Admittedly I haven't spent as much time on perfecting `Fixed32` type support as I have with `float` and `double`. One thing I noticed early on with `Fixed32` is that it doesn't really have enough numerical fidelity to be suitable for the code (even though byte-wise it's the same size as a `float`). I'm not sure what to do about that generally speaking but removing `Fixed32` as a documented option has been on my mind.

In any case, I'm going to look now into this bug you've found. I believe I can get it working for HelloWorld at least. I'll try and also update the comment documentation to better recognize limitations of `Fixed32`.

**Comment by Marreck** [comment]
Thanks for the clarification!

I wanted to check if the performance and precision costs were worth it for `Fixed32` compared to `float`. So do you think the precision would be too low to be of any use? Would `Fixed64` be usable in this regard? (I also noted this type is disabled on windows) 

I am asking because I think a fixed point physics simulation would be really helpful in some scenarios and this is something that isn't really available at the moment.

**Comment by louis-langholtz** [comment]
Just FYI: I don't think you'll find very good performance with my fixed point types implementation. When I wrote the `Fixed` point code I was more interested in proof-of-concept and seeing whether it'd help reveal any of the inaccuracies of floating point arithmetic than in performance.

**Comment by louis-langholtz** [comment]
TL;DR: Fixed now with most recent push.

Details...

I was able to recreate the occurrence of the `assert` in running the Hello World application with the real number type set to `Fixed32`. Fixing it though was less easy than I'd hoped. I wound up decreasing `Fixed32`'s fraction bits (from 14 down to 9) in order to give it enough whole bits to avoid overflowing (resulting in asserts).

As of commit 22ee1cdaa614a7cb7e38a2de58ec9a9d8e62e86a, I've pushed changes that include tweaks of `Fixed32` to get it to work for this application. I wouldn't recommend using `Fixed32` though. I'd even recommend against using it unless it's for a very limited application like the Hello World one where the simulation is completely repeatable and you've seen it work properly.

If you really want to use fixed point arithmetic, I'd recommend using the `Fixed64` type over the `Fixed32` type if possible. Unfortunately, that's not implemented for Windows (which doesn't seem to support a 128-bit integer at least using MSVS C++).

**Comment by louis-langholtz** [comment]
I'm gonna close this issue now but go ahead and add to it and/or re-open it the problem still occurs.
